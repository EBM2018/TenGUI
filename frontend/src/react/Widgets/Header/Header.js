import React from 'react';
import { instanceOf } from 'prop-types';
import {Cookies, withCookies} from 'react-cookie';
import UserAccountWidget from '../UserAccountWidget/UserAccountWidget';

import logoAndName from '../../../images/LogoTenuGUI.png';

class Header extends React.PureComponent {
        static propTypes = {
          cookies: instanceOf(Cookies).isRequired,
        }

        render() {
          const { cookies } = this.props;
          const userJSON = cookies.get('userJSON');
          if (userJSON !== undefined) {
            return (
              <header>
                <div className="columns add-margin">
                  <div className="column" />
                </div>
                <div className="columns add-margin">

                  <div className="column is-2">
                    <figure className="image">
                      <img src={logoAndName} alt="TenuGUI" />
                    </figure>
                  </div>

                  <div className="column is-8" />
                  <div className="column">
                    <UserAccountWidget />
                  </div>
                </div>
              </header>
            );
          }
          return <></>;
        }
}

export default withCookies(Header);
