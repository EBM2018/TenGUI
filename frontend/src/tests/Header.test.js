import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../react/Widgets/Header/Header';


test('does this render', () => {
  const component = renderer
    .create(<Header />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
