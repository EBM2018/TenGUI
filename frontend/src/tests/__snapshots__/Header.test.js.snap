// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does this render 1`] = `
<a>
  Header
</a>
`;
